src/xcode/ENA/ENA/Source/Scenes/HealthCertificates/Overview/Cells/HealthCertifiedPerson/HealthCertifiedPersonCellModel.swift

```diff
@@ -62,6 +62,19 @@ class HealthCertifiedPersonCellModel {
 		}
 
 		self.onTapToDelete = nil
+		
+		func shouldShowCaptionFor(initialCertificate certificate: HealthCertificate) -> Bool {
```

Comment:
Please move this function outside of the init and make it private.

Reply:
The reason I used an `inner func` is, that the logic I replaced with this PR get wrapped in a naming function, that encapsulate the code and make it more readable.
Unfortunately you cannot use a function (as a `private` one in a constructor), before the initialization of stored properties like `self.caption` is done.
You would get that compiler error:
```swift
'self' used in method call 'shouldShowCaptionFor' before all stored properties are initialized
```

As an alternative we could implement that function as an `extension` of `HealthCertificate`.
But that logic is view dependent and not globally valid. That's why I implement it in the regarding ViewModel class.

Reply:
I had with Omer and we recommend if you could just use the code with out the function in else block since it is just called once :)

Reply:
exactly so just extract the code block from the function and use it directly if that is ok with you 👍

Reply:
okay, the inner func is now removed. 👍